### PR TITLE
0.57.1

### PR DESCRIPTION
**Pull request by JamesRobertsonGames**
[PKG-2382]

[PKG-2382]: https://anaconda.atlassian.net/browse/PKG-2382?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- Removed non-neccessary dependencies
- Locked versions
- Update to 0.57.1

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -40,11 +42,11 @@ requirements:
     - llvmlite 0.40.0
     - numpy {{ numpy_min_ver }}
     - tbb-devel 2021.8.0
-    - _openmp_mutex            # [linux]
-    - importlib_metadata       # [py<39]
+    - _openmp_mutex 5.1        # [linux]
+    - importlib_metadata 6.0.0 # [py<39]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - importlib_metadata  # [py<39]
```
For python packages in host, we can drop pinnings

File: recipe/meta.yaml [hunk 2]
```diff
@@ -40,11 +42,11 @@ requirements:
...
   run:
     - python
-    - importlib_metadata       # [py<39]
+    - importlib_metadata 6.0.0 # [py<39]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - importlib_metadata  # [py<39]
```
See https://github.com/numba/numba/blob/04e81073b2c1e3ff4afa1da8513738e5e136775b/setup.py#L372

File: recipe/meta.yaml [hunk 3]
```diff
@@ -53,8 +55,6 @@ requirements:
     # xref: https://github.com/numba/numba/issues/8529
     - numpy >={{ numpy_min_ver }},!=1.22.0,!=1.22.1,!=1.22.2,<1.25
     # needed for pkg_resources
-    - setuptools
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:56]: Yeah, looks like it was really removed from the source code on August 18, 2022 https://github.com/numba/numba/pull/8366/files

File: recipe/meta.yaml (hunk 1 above)
Comment (JamesRobertsonGames) [reply recipe/meta.yaml]: Does it not matter that this will trigger a linter error?

**Comment by lorepirri** [comment]
regarding:
```
{% set numpy_min_ver = "1.21" %}  # [py<311]
{% set numpy_min_ver = "1.23" %}  # [py==311]
```

with

```
numpy {{ numpy_min_ver }}
```

shall we use our:

```
numpy {{ numpy }}
```

from the `aggregate/conda_build_config.yaml` or this one is an exception?

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml]: I also don't get it. In the recipe the version is not specified, but this is often the case. @skupr-anaconda why shall we remove this one?

**Review by lorepirri — APPROVED** [review]
LGTM, the rest is nitpick

**Comment by JamesRobertsonGames** [comment]
> regarding:
> 
> ```
> {% set numpy_min_ver = "1.21" %}  # [py<311]
> {% set numpy_min_ver = "1.23" %}  # [py==311]
> ```
> 
> with
> 
> ```
> numpy {{ numpy_min_ver }}
> ```
> 
> shall we use our:
> 
> ```
> numpy {{ numpy }}
> ```
> 
> from the `aggregate/conda_build_config.yaml` or this one is an exception?

@lorepirri this seems to be an exception as numba links with specific numpy version ranges, and is unknown whether or not it will support a different version going forward

**Comment by lorepirri** [comment]
> > regarding:
> > ```
> > {% set numpy_min_ver = "1.21" %}  # [py<311]
> > {% set numpy_min_ver = "1.23" %}  # [py==311]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > with
> > ```
> > numpy {{ numpy_min_ver }}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > shall we use our:
> > ```
> > numpy {{ numpy }}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > from the `aggregate/conda_build_config.yaml` or this one is an exception?
> 
> @lorepirri this seems to be an exception as numba links with specific numpy version ranges, and is unknown whether or not it will support a different version going forward

@JamesRobertsonGames ok that's cool! thanks!

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Comment by lorepirri** [comment]
I've deleted the branch and add the automation in the settings

**Pull request merged**